pkg/cloud/scope/shared.go

```diff
@@ -127,3 +132,23 @@ func (p *defaultSubnetPlacementStrategy) getSubnetsForAZs(azs []string, controlP
 
 	return subnetIDs, nil
 }
+
+// getUnstructuredControlPlane returns the unstructured object for the control plane, if any.
+func getUnstructuredControlPlane(ctx context.Context, client client.Client, cluster *clusterv1.Cluster) (*unstructured.Unstructured, error) {
+	if cluster.Spec.ControlPlaneRef == nil {
+		// If the control plane ref is not set, return an empty object.
+		// Not having a control plane ref is valid given API contracts.
+		return &unstructured.Unstructured{}, nil
```

Comment:
If we never use `nil`, do we need to return a pointer?

Reply:
The only downside about not having a pointer is the implicit when passing it around?

Reply:
I guess you meant implicit copy. Fair point.